File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    nums_people = len(people)
    nums_pair = 0

    start = 0
    end = nums_people - 1

    while start < end:
        if people[start] + people[end] <= limit:
            nums_pair += 1
            start += 1
        end -= 1

    return nums_people - nums_pair

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;


class Solution {
    
    public int solution(int[] people, int limit) {
        Arrays.sort(people);
        int numberOfPeople = people.length;
        int numberOfPair = 0;

        int start = 0;
        int end = numberOfPeople - 1;

        while (start < end) {
            int weight = people[start] + people[end];

            if (weight <= limit) {
                numberOfPair++;
                start++;
            } 
            end--;
        }
        return numberOfPeople - numberOfPair;
    }
}